test/api_container_create_test.go

```diff
@@ -123,3 +118,115 @@ func (suite *APIContainerCreateSuite) TestNonExistingImg(c *check.C) {
 	c.Assert(err, check.IsNil)
 	CheckRespStatus(c, resp, 404)
 }
+
+// TestBadParam tests using bad parameter return 400.
+func (suite *APIContainerCreateSuite) TestBadParam(c *check.C) {
+
+	// TODO:
+	// 1. Invalid container name, for example length too large, illegal letter.
+	// 2. Invalid Parameters
+}
+
+// TestHostConfigNil tests using bad parameter return 500.
+func (suite *APIContainerCreateSuite) TestHostConfigNil(c *check.C) {
+	cname := "TestHostConfigNil"
+	q := url.Values{}
+	q.Add("name", cname)
+	query := request.WithQuery(q)
+
+	// HostConfig is nil.
+	obj := map[string]interface{}{
+		"Image":      busyboxImage,
+		"HostConfig": nil,
+	}
+
+	path := "/containers/create"
+	body := request.WithJSONBody(obj)
+
+	resp, err := request.Post(path, query, body)
```

Comment:
Could we use `resp, err := request.Post("/containers/create", query, body)` directly, like we discussed before.

If that is OK, maybe in a follow-up, we can change all of those in one commit.

Reply:
ok, I'll change once I met